### PR TITLE
sol-aio: Add C API to handle Analog I/O (V4)

### PR DESCRIPTION
**Pull request by mbelluzzo**
Applied changes suggested by @lpereira

### CODE REVIEW

File: src/lib/io/sol-aio-linux.c [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+{
+    char path[PATH_MAX];
+
+    if (!AIO_PATH(path, aio->device, aio->pin))
+        return false;
+
+    aio->fp = fopen(path, "re");
+    if (!aio->fp)
+        return false;
+    setvbuf(aio->fp, NULL, _IONBF, 0);
```
Comment (edersondisouza) [inline src/lib/io/sol-aio-linux.c]: Why is this needed?

Comment (mbelluzzo) [inline src/lib/io/sol-aio-linux.c]: Because there is no need for buffering in this case.

**Comment by glima** [comment]
Other than that, LGTM

File: src/lib/io/sol-aio-linux.c (hunk 1 above)
Comment (edersondisouza) [inline src/lib/io/sol-aio-linux.c]: Sounds fair, but on iio isn't the buffer located on `/dev/iio:deviceX`? I mean, is this really necessary? (If so, I may need to change something on my pr #441)

Comment (edersondisouza) [inline src/lib/io/sol-aio-linux.c]: Or is this not related to the iio buffers?

Comment (ibriano) [inline src/lib/io/sol-aio-linux.c]: On Wed, 05 Aug 2015 12:34:30 -0700, Ederson de Souza wrote:
> > +    int pin;
> > +    unsigned int mask;
> > +};
> > +
> > +static bool
> > +_aio_open_fp(struct sol_aio *aio)
> > +{
> > +    char path[PATH_MAX];
> > +
> > +    if (!AIO_PATH(path, aio->device, aio->pin))
> > +        return false;
> > +
> > +    aio->fp = fopen(path, "re");
> > +    if (!aio->fp)
> > +        return false;
> > +    setvbuf(aio->fp, NULL, _IONBF, 0);
> 
> Or is this not related to the iio buffers?
> 

No, it's the internal buffer in FILE*

**Comment by mbelluzzo** [comment]
Inc. V5 with power on/off management.
